### ISSUE TITLE
Retry request in interceptors response

### ISSUE THREAD

**Issue by nkovacic** [ava/use-http#170]
Usually servers handles token expiration and then with 401 response tells the client to refresh token.
In the examples this is handled in request interceptor, but is there a way to handle it response and retry all failed requests?

**Comment by alex-cory** [comment]
The feature that you're looking for is on my todos
![image](https://user-images.githubusercontent.com/5455859/73526918-046c9800-43c7-11ea-8833-ea667f07d7eb.png)

If you don't require having use-http manage all of your state, you can do
```js
const attempts = useRef(3)
const [request, response] = useFetch({ path: '/whatever' })

const getUser = async () => {
  let data = await request.get()
  if (response.status === 401 && attempts.current > 0) {
    attempts.current--
    return await getUser()
  }
  return data
}

```

**Comment by alex-cory** [comment]
@nkovacic does this help at all?

**Comment by nkovacic** [comment]
It could work this way, but currently opted for the client side check for token expiration, because currently its the only ways to do this in one place.

**Comment by alex-cory** [comment]
I will try to get this implemented as soon as I can!

**Comment by alex-cory** [comment]
This feature is coming very soon! See [this PR](https://github.com/alex-cory/use-http/pull/218)

**Comment by nkovacic** [comment]
Great, thank you for your effort 👍

**Comment by alex-cory** [comment]
@nkovacic This feature is implemented in `v0.4.5`
